src/k8s/pkg/utils/certificate.go

```diff
@@ -26,3 +31,58 @@ func SplitIPAndDNSSANs(extraSANs []string) ([]net.IP, []string) {
 
 	return ipSANs, dnsSANs
 }
+
+func TLSClientConfigWithTrustedCertificate(remoteCert *x509.Certificate) (*tls.Config, error) {
+	if remoteCert == nil {
+		return nil, fmt.Errorf("invalid remote public key")
+	}
+
+	config := &tls.Config{}
+
+	remoteCert.IsCA = true
+	config.RootCAs.AddCert(remoteCert)
+
+	// Always use public key DNS name rather than server cert, so that it matches.
+	if len(remoteCert.DNSNames) > 0 {
+		config.ServerName = remoteCert.DNSNames[0]
```

Comment:
So basically, we use the `DNSName` of the CA as the server name of the clients TLS config. What happens if there is no `DNSName` set? Will it break if there is no server name set for the config?

Currently, we have our own CA.
What happens if the CA comes from an external vendor (e.g. Vault)? Will this still work?

Reply:
@neoaggelos What do you think?

Reply:
Resolved offline: tldr: here CA = remote certificate, the kubernetes CA is not involved at all.